client/dfclient/include/text_creator.hpp

```diff
@@ -0,0 +1,118 @@
+	float _last_y;
+	float _lastScale;
+	int _last_from, _last_hold, _last_decay;
+
+	// existent for code clarity
+	inline void setColor(ncine::Color color) { _textColor = color; }
+	inline void setColor(int r, int g, int b, int a = 255) { _textColor = ncine::Color(r,g,b,a); }
+
+	inline void setPosition(float x, float y) { _pos_x = x; _pos_y = y; }
+	inline void setScale(float scale) { _textScale = scale; }
+	inline void setFrom(int from) { _from = from; }
+	inline void setHold(int hold) { _hold = hold; }
+	inline void setDecay(int decay) { _decay = decay; }
+	inline void setTweenParams(int from, int hold, int decay) { setFrom(from); setHold(hold); setDecay(decay); }
+	inline void setLayer(Layers layer) { _layer = layer; }
+
+	/** Create an ncine::TextNode and (by default, but optionally) an alpha fadeout tween. 
+	 * 
+	 * The main function of the TextCreator. Returns a unique pointer to 
+	 * a newly created ncine::TextNode and, unless stated otherwise via 
+	 * parameters, creates an alpha fadeout tween, placing it in 
+	 * _resources->_intTweens. 
+	 * 
+	 * The alpha fadeout tween starts at {from}, holds the initial value 
+	 * for {hold} frames and then goes down to 0 during {decay} frames, 
+	 * optionally following a function defined by {easing}. Specifying 
+	 * {decay} as < 0 skips the tween creation, making the text permanently 
+	 * visible unless later modified. 
+	 * 
+	 * Every parameter (besides text) can be left unspecified via omission or 
+	 * std::nullopt to use previously set defaults. 
+	 * 
+	 * Texts created are scaled with regard to window width.
+	 * 
+	 * @param text The displayed text value. The only mandatory argument.
+	 * @param color The color of the text. Not including alpha, which is set 
+	 * 	via <from>.
+	 * @param pos_x The x position of the text center, measured in pixels.
+	 * @param pos_y The y posiion of the text center, measured in pixels.
+	 * @param scale Scaling factor (1.0f is default value).
+	 * @param from The starting alpha value (0-255).
+	 * @param hold The number of frames to hold the initial alpha value.
+	 * @param decay The number of frames to take to reduce alpha to 0. Set 
+	 * 	as < 0 to skip tween creation.
+	 * @param layer A layer from Layers enum to draw the text on.
+	 * @param easing The tween easing function. For more details, see 
+	 * 	the tweeny::via() definition.
+	 * @return An std::unique_ptr< ncine::TextNode > of the created text.
+	 */
+	template <typename T = tweeny::easing::linearEasing>
+	std::unique_ptr<ncine::TextNode> CreateText(
+		std::string text,
+		std::optional<ncine::Color> color = std::nullopt,
+		std::optional<float> pos_x = std::nullopt,
+		std::optional<float> pos_y = std::nullopt,
+		std::optional<float> scale = std::nullopt,
+		std::optional<int> from = std::nullopt,
+		std::optional<int> hold = std::nullopt,
+		std::optional<int> decay = std::nullopt,
+		std::optional<Layers> layer = std::nullopt,
+		const T& easing = tweeny::easing::linear
+		);
```

Comment:
Well, I wanted to make this tool as universal as can be. If you want to, you can do exactly as you have described above. I just think this:
```
textCreator->CreateText(
	"you ended " + gameData.players[deathReport->victim()].name + "'s killing spree!",
	ncine::Color(255, 70, 0),
	textsXPos[1], textsYPos[2],
	1.5f,
	255, 60, 120
);
```
is more clear and readable than this:
```
TextCreator textCreator;
textCreator.setColor(ncine::Color(255, 70, 0));
textCreator.setPosition(textsXPos[1], textsYPos[2]);
textCreator.setScale(1.5f);
textCreator.setTweenParams(255, 60, 120);
textCreator->CreateText("you ended " + gameData.players[deathReport->victim()].name + "'s killing spree!");
```

Of course, the second version is very descriptive for someone who doesn't know the tool, but the point is, the function is used very frequently, at least in this part of code. When you get to know what the parameters are for, it becomes very clear, and all the `textCreator.setX(` part becomes obsolete and just obscures the code.

Regarding your example, you don't need to `std::nullopt` your way all the way to layer, that's what the setter functions are for. You can just do:
```
textCreator.setLayer(Layers::someLayer);
textCreator.CreateText("stuff");
```
Or any mix of the two approaches.

Having said all this, I think it should stay the way it is, especially because of the support for different use styles. But you're welcome to prove me wrong :P

Reply:
I'm fine with leaving it as it is.